blocklylib-core/src/main/java/com/google/blockly/android/ui/fieldview/FieldAdapter.java

```diff
@@ -0,0 +1,42 @@
+package com.google.blockly.android.ui.fieldview;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.AppCompatSpinner;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.List;
+
+class FieldAdapter<T> extends ArrayAdapter<T> {
+    private AppCompatSpinner mSpinner;
+
+    public FieldAdapter(@NonNull Context context, int resource, @NonNull List<T> objects, @Nullable AppCompatSpinner mSpinner) {
+        super(context, resource, objects);
+        this.mSpinner = mSpinner;
+    }
+
+    public FieldAdapter(@NonNull Context context, int resource, @Nullable AppCompatSpinner mSpinner) {
+        super(context, resource);
+        this.mSpinner = mSpinner;
+    }
+
+    @NonNull
+    @Override
+    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+        // Generate Stack Trace
+        StringWriter stackTrace = new StringWriter();
+        PrintWriter printer = new PrintWriter(stackTrace);
+        new RuntimeException().printStackTrace(printer);
+        // Use Stack Trace To Check if Method is being Called By android.widget.Spinner.measureContentWidth (fixes #735)
+        if (mSpinner != null && stackTrace.toString().contains("at android.widget.Spinner.measureContentWidth")) {
+            return super.getView(mSpinner.getSelectedItemPosition(), convertView, parent);
```

Comment:
This is a very heavyweight fix. The creation, serialization, and search of the stack trace should not be done lightly.  Is there any external conversation about spinner bugs that lead you to this solution?

How often is `getView(..)` called? Every visible item in the list for every relayout (which probably/usually isn't too much).

My gut says there must be a simpler solution, and that the above solution really hints at some misuse of some Spinner API in the current Blockly code.

Reply:
If it always does the first branch in the ```if```, you can't change the currently selected variable in variable drop-downs. You can't set a property in the Spinner object when calling onMeasure (via overriding the method) because for some reason, the property may sometimes be ```false```, when ```getView``` even though it is set to ```true``` before the ```super``` method is called. is called. Unless there is an easy way to determine if a method is called from ```onMeasure```, I can't think of any other solution.

Reply:
I created a test app to see how much longer the stack trace made   ```onMeasure```. The app ran ```measure``` 1000 times on a ```Spinner``` that had the stack trace serialization and one that didn't and then shows the results. On the emulator the difference between using and not using the stack trace seems to be insignificant.

![screenshot_1547081657](https://user-images.githubusercontent.com/17478432/50939040-28221080-1449-11e9-9e61-d18873a5654c.png)

Source Code: https://github.com/TheBrokenRail/StackTraceTest

(All numbers are in milliseconds)

Reply:
As it is only 18% slower to use the stack trace, and these numbers are already in sub-milliseconds on average, for a method that won't get called much, I would say that the performance decrease is acceptable.

Reply:
Since this isn't slowing it down enough to be noticeable I'm going to resolve this conversation.